core/trino-main/src/test/java/io/trino/sql/planner/assertions/ExpressionMatcher.java

```diff
@@ -85,7 +85,7 @@ public Optional<Symbol> getAssignedSymbol(PlanNode node, Session session, Metada
         }
 
         List<Expression> matches = matchesBuilder.build();
-        checkState(matches.size() < 2, "Ambiguous expression %s matches multiple assignments", expression,
+        checkState(matches.size() < 2, "Ambiguous expression %s matches multiple assignments: %s", expression,
                 (matches.stream().map(Expression::toString).collect(Collectors.joining(", "))));
```

Comment:
replace second arg with just `matches`

- it's sufficient, list toString will be nice
- checkState formatting params should be cheap to evaluate, or one should use an `if`

Reply:
as separate commit